client/python/armada_client/client.py

```diff
@@ -128,6 +129,32 @@ def cancel_jobs(
         response = self.submit_stub.CancelJobs(request)
         return response
 
+    def cancel_jobset(
+        self,
+        queue: str,
+        job_set_id: str,
+        filter_states: List[JobState],
+    ) -> empty_pb2.Empty:
+        """Cancel jobs in a given queue.
+
+        Uses the CancelJobSet RPC to cancel jobs.
+        A filter is used to only cancel jobs in certain states.
+
+        :param queue: The name of the queue
+        :param job_set_id: An array of JobSubmitRequestItems.
+        :param filter_states: A list of states to filter by.
+        :return: An empty response.
+        """
+
+        job_filter = submit_pb2.JobSetFilter(
+            states=(state.value for state in filter_states)
```

Comment:
This creates a generator. It's almost certainly not what you want here. You probably want to make it a list immediately, right? 

```
>>> lol = (x for x in [1,2,3])
>>> dir(lol)
['__class__', '__del__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__lt__', '__name__', '__ne__', '__new__', '__next__', '__qualname__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'close', 'gi_code', 'gi_frame', 'gi_running', 'gi_yieldfrom', 'send', 'throw']
>>> lol
<generator object <genexpr> at 0x7f0685be8270>
>>> next(lol)
1
>>> next(lol)
2
>>> next(lol)
3
>>> next(lol)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```

Reply:
Yeah good spot. Will fix that